code/modules/clothing/masks/mystic.dm

```diff
@@ -1,75 +1,164 @@
+		to_chat(M, "<span class=sinister>Go away you dumb animal, we have no interest in you.</span>")
+		M.visible_message("<span class=warning>\The [src] slips right off of [M]'s face.</span>")
+		M.drop_from_inventory(src)
 		return
+
 	if(H.wear_mask == src)
 		flick("happiest_flash", src)
-		to_chat(H, "<span class='sinister'>Your thoughts are bombarded by incessant laughter.</span>")
+		to_chat(H, "<span class='warning'>Your thoughts are bombarded by incessant laughter.</span>\n<span class='sinister'>Oh joy! [H.real_name]'s decided to join the party!</span>")
 		H << sound('sound/effects/hellclown.ogg')
 		canremove = 0
 
 /obj/item/clothing/mask/happy/attack_hand(mob/user as mob)
 	if(user.wear_mask == src)
-		to_chat(user, "<span class='sinister'>It won't come off.</span>")
 		flick("happiest_flash", src)
+		to_chat(user, "<span class='sinister'>Why would you want to get rid of us, aren't you having fun?</span>")
+		changehappiness(UNHAPPIER)
 	else
 		..()
 
 /obj/item/clothing/mask/happy/pickup(mob/user as mob)
 	flick("happiest_flash", src)
-	to_chat(user, "<span class=warning><B>The mask's eyesockets briefly flash with a foreboding red glare.</span></B>")
+	to_chat(user, "<span class='warning'><B>The mask's eyesockets briefly flash with a foreboding red glare.</span></B>\n<span class='sinister'>We're just so very happy to have you joining us. Bring us just a little closer.</span>")
 
-/obj/item/clothing/mask/happy/OnMobLife(var/mob/living/carbon/human/wearer)
-	var/mob/living/carbon/human/W = wearer
-	if(W.wear_mask == src)
-		RaiseShade(W)
-		if(prob(5))
-			switch(pick(1,2,3))
-				if(1)
-					W.say(pick("I'M SO HAPPY!", "SMILE!", "ISN'T EVERYTHING SO WONDERFUL?", "EVERYONE SHOULD SMILE!"))
-				if(2)
-					var/list/laughtypes = list("funny", "disturbing", "creepy", "horrid", "bloodcurdling", "freaky", "scary", "childish", "deranged", "airy", "snorting")
-					var/laughtype = pick(laughtypes)
-					W.visible_message("[W] makes \a [laughtype] laugh.")
-				if(3)
-					W.emote(pick("laugh", "chuckle", "giggle", "grin", "smile"))
+/obj/item/clothing/mask/happy/OnMobLife(var/mob/living/carbon/human/W)
+	if(istype(W) && W.wear_mask == src)
+		if(happiness <= 0)
+			flick("happiest_flash", src)
+			to_chat(W, "<span class='sinister'>It seems you're not being a very good friend to us. We'll just have to fire up this relationship!</span>")
+			var/datum/organ/external/affecting = W.get_organ(LIMB_HEAD)
+			if(affecting.take_damage(0, 20))
+				W.UpdateDamageIcon(1)
+			laugh(W)
+			changehappiness(-UNHAPPIER)
+
+		else if(happiness >= VERYHAPPY)
+			var/bruted = W.getBruteLoss()
+			var/burned = W.getFireLoss()
+			if(bruted + burned > 0 && prob(happiness))
+				var/healed = 0
+				for(var/mob/living/carbon/human/H in view(5)) //the mask only draws from other humans
+					var/heal = 0
+					if(H == W || H.stat == DEAD)
+						continue
+
+					if(bruted > 0) //only heal one damage type per mob, preferring brute
+						heal = min(5, bruted)
```

Comment:
This seems pretty nuts. Basically, if you pulse cannon someone down to crit in the span of one tick, and you're the only mob in view, the mask might instacrit you to restore them to max health. Well, not just might -- since it's Very Happy there's a 50% chance of this happening every single life tick.